### PR TITLE
add an experimental 'flutter build web' call to the backend

### PR DESCRIPTION
**Pull request by devoncarew**
Add an experimental 'flutter build web' call to the backend:

- add an experimental `api/v2/_flutterBuild` call to the backend
- it's given a source file and invokes `flutter build web`; this is currently slower than we'd want but I believe there's some room for improvement in the execution time of `flutter build web`
- the flutter web frontend has been updated to use this compile call for both dart and flutter snippets

Many of the changes here are multiple copies of the proto related files. With this change, the new front-end can now run flutter snippets.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Comment by devoncarew** [comment]
Screenshot 2023-07-18 at 10 41 43 AM

File: pkgs/sketch_pad/README.md [hunk 1]
```diff
@@ -16,5 +16,5 @@ This is an experimental re-imagining of the DartPad UI. Goals include:
 To run this locally, run:
 
 ```
-flutter run -d chrome
+flutter run -d chrome --web-browser-flag "--disable-web-security"
```
Comment (johnpryan) [inline pkgs/sketch_pad/README.md:19]: What does this flag do?

File: pkgs/sketch_pad/lib/model.dart [hunk 2]
```diff
@@ -186,10 +186,19 @@ class AppServices {
...
 
-  Future<CompileResponse> compile(CompileRequest request) async {
+  // Future<CompileResponse> compile(CompileRequest request) async {
+  //   try {
+  //     appModel.compilingBusy.value = true;
+  //     return await services.compile(request);
+  //   } finally {
+  //     appModel.compilingBusy.value = false;
+  //   }
+  // }
```
Comment (johnpryan) [inline pkgs/sketch_pad/lib/model.dart:196]: nit: commented out code

**Review by johnpryan — APPROVED** [review]
Is this still in the draft stage? Or would you prefer to merge it into main?

File: pkgs/sketch_pad/lib/model.dart (hunk 2 above)
Comment (devoncarew) [reply pkgs/sketch_pad/lib/model.dart]: will update this

File: pkgs/sketch_pad/README.md (hunk 1 above)
Comment (devoncarew) [reply pkgs/sketch_pad/README.md:19]: This flag tells chrome to not require that the server use the correct CORS headers for loading some of the flutter web resources (`AssetManifest.json`, `FontManifest.json`, ...). We could probably also address this is the web server used by `flutter run -d chrome` had some config options that would enable us to set CORS headers for some files.

In any case, I'll document this in the readme.

File: pkgs/dart_services/lib/src/compiler.dart [hunk 3]
```diff
@@ -240,6 +240,67 @@ class Compiler {
...
+      sourceFile.writeAsStringSync(source);
+
+      final arguments = <String>[
+        'build',
+        'web',
+
+        // This disables minification.
+        '--dart2js-optimization=O1',
+
+        // With the web renderer, we don't need to load other (skiawasm) resources.
```
Comment (devoncarew) [inline pkgs/dart_services/lib/src/compiler.dart:271]: @kevmoo has asked that we look into instead using the skiawasm backend (likely loaded from a CDN). I'll add a todo.

**Review by devoncarew — COMMENTED** [review]
> Is this still in the draft stage? Or would you prefer to merge it into main?

I'd prefer to merge it - it'll make it easier to iterate on (the restful API is named w/ an underscore - `_flutterBuild` - to help make sure we communicate 'do not use' to people).

**Pull request merged**